corehq/apps/users/bulk_download.py

```diff
@@ -163,6 +164,41 @@ def count_users_and_groups(domain, user_filters, group_memoizer):
     return users_count + groups_count
 
 
+def dump_usernames(domain, download_id, user_filters, task):
+    writer = Excel2007ExportWriter(format_as_text=True)
+    users_count = get_commcare_users_by_filters(domain, user_filters, count_only=True)
+    DownloadBase.set_progress(task, 0, users_count)
+
+    role_id = user_filters.get('role_id', None)
+    search_string = user_filters.get('search_string', None)
+    location_id = user_filters.get('location_id', None)
+
+    query = UserES().domain(domain).mobile_users()
+
+    if role_id:
+        query = query.role_id(role_id)
+    if search_string:
+        query = query.search_string_query(search_string, default_fields=['first_name', 'last_name', 'username'])
+    if location_id:
+        location_ids = SQLLocation.objects.get_locations_and_children_ids([location_id])
+        query = query.location(location_ids)
+
+    usernames = query.values_list('username', flat=True)
+
+    use_transfer = settings.SHARED_DRIVE_CONF.transfer_enabled
+    filename = "{}_users_{}.xlsx".format(domain, uuid.uuid4().hex)
```

Comment:
Realize you copied this from elsewhere, but that uuid doesn't seem helpful. It'd be more meaningful to users to include the domain name and maybe the other filters.

Reply:
i am fine with removing the uuid.
Adding filters would add more logic to this like fetching the location to get its name, location id in itself might not be useful, which i would like to avoid? I think adding a timestamp can be helpful to uniquely identify the file but that can anyway be viewed by the user when seeing the file on their computers anyway.

Reply:
Updated.

Reply:
Cool, and I agree filters aren't absolutely necessary and timestamp wouldn't add anything.